sklearn/model_selection/_split.py

```diff
@@ -1708,3 +1714,23 @@ def _build_repr(self):
         params[key] = value
 
     return '%s(%s)' % (class_name, _pprint(params, offset=len(class_name)))
+
+
+def _check_X_y_groups(X, y, groups):
+    """Validate X, y and groups for CV splitters"""
+    # Be flexible with the input data dimensions and presence of nan etc.
+    # to allow any domain specific data to be split
+    X = check_array(X, accept_sparse=['coo', 'csr', 'csc'],
+                    dtype=None, force_all_finite=False, ensure_2d=False,
+                    allow_nd=True)
+    if y is not None:
+        y = check_array(y, accept_sparse=['coo', 'csr', 'csc'],
```

Comment:
Same for y. Are these tested? Should they be? I guess we should be as loose as possible with the test as long as the cross-validation classes work.